components/eam/src/physics/crm/cospsimulator_intr.F90

```diff
@@ -0,0 +1,3184 @@
+
+#ifdef USE_COSP     
+
+    use cam_history,         only: addfld, add_default, horiz_only
+#ifdef SPMD
+    use mpishorthand,        only : mpir8, mpiint, mpicom
+#endif
+    use netcdf,              only : nf90_open, nf90_inq_varid, nf90_get_var, nf90_close, nf90_nowrite
+    use error_messages,      only : handle_ncerr, alloc_err
+    
+    use physics_buffer,  only: pbuf_get_index
+
+    use mod_cosp_config,  only : R_UNDEF    
+    
+    integer :: errcode
+    integer :: i
+
+    ! Are we using MMF?
+    call phys_getopts(use_MMF_out=use_MMF)
+    
+    ! ISCCP OUTPUTS
+    if (lisccp_sim) then
+       !! addfld calls for all
+       call addfld('FISCCP1_COSP',(/'cosp_tau','cosp_prs'/),'A','percent', &
+            'Grid-box fraction covered by each ISCCP D level cloud type',&
+            flag_xyfill=.true., fill_value=R_UNDEF)
+       
+       !*cfMon,cfDa* tclisccp (time,profile), CFMIP wants "gridbox mean cloud cover from ISCCP"
+       call addfld('CLDTOT_ISCCP', horiz_only,'A','percent', &
+            'Total Cloud Fraction Calculated by the ISCCP Simulator ',flag_xyfill=.true., fill_value=R_UNDEF)
+       !*cfMon,cfDa* albisccp (time,profile)
+       ! Per CFMIP request - weight by ISCCP Total Cloud Fraction (divide by CLDTOT_ISSCP in history file to get weighted average)
+       call addfld('MEANCLDALB_ISCCP',horiz_only,'A','1','Mean cloud albedo*CLDTOT_ISCCP',flag_xyfill=.true., fill_value=R_UNDEF)
+       !*cfMon,cfDa* ctpisccp (time,profile)
+       ! Per CFMIP request - weight by ISCCP Total Cloud Fraction (divide by CLDTOT_ISSCP in history file to get weighted average)
+       call addfld('MEANPTOP_ISCCP',horiz_only,'A','Pa','Mean cloud top pressure*CLDTOT_ISCCP',flag_xyfill=.true., &
+            fill_value=R_UNDEF)
+       ! tauisccp (time,profile)
+       ! For averaging, weight by ISCCP Total Cloud Fraction (divide by CLDTOT_ISSCP in history file to get weighted average)
+       call addfld ('MEANTAU_ISCCP',horiz_only,'A','1','Mean optical thickness*CLDTOT_ISCCP',flag_xyfill=.true., &
+            fill_value=R_UNDEF)
+       ! meantbisccp (time,profile), at 10.5 um
+       call addfld ('MEANTB_ISCCP',horiz_only,'A','K','Mean Infrared Tb from ISCCP simulator',flag_xyfill=.true., &
+            fill_value=R_UNDEF)
+       ! meantbclrisccp (time,profile)
+       call addfld ('MEANTBCLR_ISCCP',horiz_only,'A','K','Mean Clear-sky Infrared Tb from ISCCP simulator',   &
+            flag_xyfill=.true., fill_value=R_UNDEF)
+       ! boxtauisccp (time,column,profile)
+       call addfld ('TAU_ISCCP',(/'cosp_scol'/),'I','1','Optical Depth in each Subcolumn',flag_xyfill=.true., fill_value=R_UNDEF)
+       ! boxptopisccp (time,column,profile)
+       call addfld ('CLDPTOP_ISCCP',(/'cosp_scol'/),'I','Pa','Cloud Top Pressure in each Subcolumn',  &
+            flag_xyfill=.true., fill_value=R_UNDEF)
+
+       !! add all isccp outputs to the history file specified by the CAM namelist variable cosp_histfile_num
```

Comment:
I see a lot of these double/triple bang comments, it seems some past developer was inconsistent with their formatting!

Reply:
Indeed.

Reply:
These should be removed now.